### PR TITLE
Unit Localization cluster server

### PR DESCRIPTION
**Pull request by lzgrablic02**
#### Problem
* No Unit Localization server implementation.
* Issues with Unit Localization cluster XML.

#### Change overview
Unit Localization server implementation.
Renamed XML file to match name in zcl.json and all.xml.
Added explicit ClusterRevision.
Corrected type for TemperatureUnit.
Added FeatureMap.
ZAP regen.

#### Testing
How was this tested? (at least one bullet point required)
* Write and read TemperatureUnit.
* Read FeatureMap.
* Read ClusterRevision.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: zzz_generated/all-clusters-app/zap-generated/endpoint_config.h [hunk 1]
```diff
@@ -59,12 +59,17 @@
             0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,      \
             0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,                                                                              \
                                                                                                                                    \
+            /* Endpoint: 0, Cluster: Unit Localization (server), big-endian */                                                     \
+                                                                                                                                   \
+            /* 284 - FeatureMap, */                                                                                                \
+            0x00, 0x00, 0x00, 0x00,                                                                                                \
```
Comment (bzbarsky-apple) [inline zzz_generated/all-clusters-app/zap-generated/endpoint_config.h]: This is not the right value, right?  This is claiming that the TEMP feature is NOT supported.

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap [hunk 2]
```diff
@@ -1764,6 +1764,72 @@
...
+            {
+              "name": "FeatureMap",
+              "code": 65532,
+              "mfgCode": null,
+              "side": "server",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "0",
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:1810]: This defaultValue does not look right if we want to claim to support the TEMP feature.

File: src/app/zap-templates/zcl/data-model/chip/unit-localization-cluster.xml [hunk 3]
```diff
@@ -30,10 +30,15 @@ limitations under the License.
     <client init="false" tick="false">true</client>
     <server init="false" tick="false">true</server>
     <description>Nodes should be expected to be deployed to any and all regions of the world. These global regions
-      may have differing preferences for the units in which values are conveying in communication to a
+      may have differing preferences for the units in which values are conveyed in communication to a
       user. As such, Nodes that visually or audibly convey measurable values to the user need a
       mechanism by which they can be configured to use a user’s preferred unit.</description>
-    <!-- Base data types -->      
-    <attribute side="server" code="0x00" define="TEMPERATURE_UNIT" type="ENUM8" writable="true" optional="true">TemperatureUnit</attribute>
+    <globalAttribute side="either" code="0xFFFD" value="1"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/unit-localization-cluster.xml:36]: I don't think the `value="1"` here makes sense.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/door-lock-app/door-lock-common/door-lock-app.zap [hunk 4]
```diff
@@ -1426,6 +1426,72 @@
...
+          "name": "Unit Localization",
+          "code": 45,
+          "mfgCode": null,
+          "define": "UNIT_LOCALIZATION_CLUSTER",
+          "side": "server",
+          "enabled": 1,
+          "commands": [],
+          "attributes": [
+            {
+              "name": "TemperatureUnit",
```
[the remark this answers is not in the document]
Comment (lzgrablic02) [reply examples/door-lock-app/door-lock-common/door-lock-app.zap:1449]: I added it wherever time format localization was, figuring if one was relevant so was the other. Removing.

File: examples/lighting-app/lighting-common/lighting-app.matter [hunk 5]
```diff
@@ -1433,6 +1445,7 @@ endpoint 0 {
   server  cluster Switch;
   server  cluster ThreadNetworkDiagnostics;
   server  cluster TimeFormatLocalization;
+  server  cluster UnitLocalization;
```
[the remark this answers is not in the document]
Comment (lzgrablic02) [reply examples/lighting-app/lighting-common/lighting-app.matter:1448]: Removing.

File: examples/lock-app/lock-common/lock-app.matter [hunk 6]
```diff
@@ -858,6 +870,7 @@ endpoint 0 {
   server  cluster SoftwareDiagnostics;
   server  cluster ThreadNetworkDiagnostics;
   server  cluster TimeFormatLocalization;
+  server  cluster UnitLocalization;
```
[the remark this answers is not in the document]
Comment (lzgrablic02) [reply examples/lock-app/lock-common/lock-app.matter:873]: Removing.

File: examples/ota-provider-app/ota-provider-common/ota-provider-app.matter [hunk 7]
```diff
@@ -398,6 +410,7 @@ endpoint 0 {
   server  cluster OtaSoftwareUpdateProvider;
   server  cluster OperationalCredentials;
   server  cluster TimeFormatLocalization;
+  server  cluster UnitLocalization;
```
[the remark this answers is not in the document]
Comment (lzgrablic02) [reply examples/ota-provider-app/ota-provider-common/ota-provider-app.matter:413]: Removing.

File: examples/ota-requestor-app/ota-requestor-common/ota-requestor-app.matter [hunk 8]
```diff
@@ -517,6 +529,7 @@ endpoint 0 {
   server  cluster OtaSoftwareUpdateRequestor;
   server  cluster OperationalCredentials;
   server  cluster TimeFormatLocalization;
+  server  cluster UnitLocalization;
```
[the remark this answers is not in the document]
Comment (lzgrablic02) [reply examples/ota-requestor-app/ota-requestor-common/ota-requestor-app.matter:532]: Removing.

File: examples/pump-app/pump-common/pump-app.matter [hunk 9]
```diff
@@ -949,6 +961,7 @@ endpoint 0 {
   server  cluster SoftwareDiagnostics;
   server  cluster ThreadNetworkDiagnostics;
   server  cluster TimeFormatLocalization;
+  server  cluster UnitLocalization;
```
[the remark this answers is not in the document]
Comment (lzgrablic02) [reply examples/pump-app/pump-common/pump-app.matter:964]: Ack

File: examples/pump-controller-app/pump-controller-common/pump-controller-app.matter [hunk 10]
```diff
@@ -1000,6 +1012,7 @@ endpoint 0 {
   server  cluster SoftwareDiagnostics;
   server  cluster ThreadNetworkDiagnostics;
   server  cluster TimeFormatLocalization;
+  server  cluster UnitLocalization;
```
[the remark this answers is not in the document]
Comment (lzgrablic02) [reply examples/pump-controller-app/pump-controller-common/pump-controller-app.matter:1015]: If pump has it, wouldn't pump controller as well?

**Comment by lzgrablic02** [comment]
Closed https://github.com/project-chip/connectedhomeip/pull/13685 since changes are included here and checks were failing there.

File: examples/window-app/common/window-app.matter [hunk 11]
```diff
@@ -887,6 +899,7 @@ endpoint 0 {
   server  cluster SoftwareDiagnostics;
   server  cluster ThreadNetworkDiagnostics;
   server  cluster TimeFormatLocalization;
+  server  cluster UnitLocalization;
```
[the remark this answers is not in the document]
Comment (lzgrablic02) [reply examples/window-app/common/window-app.matter:902]: Removing.

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap (hunk 2 above)
Comment (lzgrablic02) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:1810]: I was thinking it gets set when TemperatureUnit is first set. Changed.

File: zzz_generated/all-clusters-app/zap-generated/endpoint_config.h (hunk 1 above)
Comment (lzgrablic02) [reply zzz_generated/all-clusters-app/zap-generated/endpoint_config.h]: Changed.

File: src/app/zap-templates/zcl/data-model/chip/unit-localization-cluster.xml (hunk 3 above)
Comment (lzgrablic02) [reply src/app/zap-templates/zcl/data-model/chip/unit-localization-cluster.xml:36]: It's the ClusterRevision

File: examples/pump-controller-app/pump-controller-common/pump-controller-app.matter (hunk 10 above)
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply examples/pump-controller-app/pump-controller-common/pump-controller-app.matter:1015]: Yes, if we keep it for pump we might as well keep it here.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: general cleanup, sufficient time passed for cross-timezone review. Delta is small (except repeated due to duplicated code).

**Pull request merged**